src/jit/morph.cpp

```diff
@@ -83,6 +83,18 @@ GenTreePtr          Compiler::fgMorphIntoHelperCall(GenTreePtr      tree,
     tree->gtCall.gtEntryPoint.addr = nullptr;
 #endif
 
+#if defined(_TARGET_X86_) && !defined(LEGACY_BACKEND)
+    if (varTypeIsLong(tree))
+    {
+        GenTreeCall* callNode = tree->AsCall();
+        ReturnTypeDesc* retTypeDesc = callNode->GetReturnTypeDesc();
+        retTypeDesc->Reset();
+        retTypeDesc->Initialize(this, callNode->gtRetClsHnd);
+        
+        NYI("Helper with TYP_LONG return type");
+    }
+#endif
+
```

Comment:
There is an issue with the parameters to helper calls where sometimes (in about 5 tests), we try to decompose the parameters twice. I'm not sure why it's happening, but we get into DecomposeNode with a GT_LONG node. I tried simply ignoring GT_LONGs in DecomposeNode, but that lead to another failure, so I just left the NYI here. For the rest, we still need to initialize the retTypeDesc when we morph the node into a call, so that when we figure out what registers to use in genMultiRegCallStoreToLocal, it has been initialized.